tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,635 @@
+                value = pd.DataFrame(value).add_prefix(f"{key}_")
+                dfs.append(value)
+            else:
+                raise ValueError("sequence's dimension should be 1d or 2d")
+
+    return dfs, dicts
+
+
+#computation for values that are not arrays/dataframes/series
+#these are collected into dictionary and processed with numpy meshgrid
+def grid_computation_dict(dicts):
+    #actual computation
+    if len(dicts) == 1:
+        key = list(dicts.keys())[0]
+        value = list(dicts.values())[0]
+        final = pd.DataFrame(value, columns=[key])
+    else:
+        res = np.meshgrid(*dicts.values())
+        #create structured array
+        #keeps data type of each value in the dict
+        outcome = np.core.records.fromarrays(res, names=",".join(dicts))
+        #reshape into a 1 column array
+        #using the size of any of the arrays obtained from the meshgrid computation
+        outcome = np.reshape(outcome, (np.size(res[0]), 1))
+        #flatten structured array into 1d array
+        outcome = np.concatenate(outcome)
+        #sort array
+        outcome.sort(axis=0, order=list(dicts))
+        #create dataframe
+        final = pd.DataFrame.from_records(outcome)
+    return final
+
+
+#this is for dataframes/series
+#this should be the final output if there are lists or lists and dicts
+#returned from check instance
+def compute_two_dfs(df1, df2):
+    #get lengths of dataframes(number of rows) and swap
+    # essentially we'll pair one dataframe with the other's length:
+    lengths = reversed([ent.index.size for ent in (df1, df2)])
+    #grab the maximum string length
+    string_cols = [
+        frame.select_dtypes(include="object").columns for frame in (df1, df2)
+    ]
+
+    #pair max string length with col
+    #will be passed into frame.to_records, to get dtype in numpy recarray
+    string_cols = [{col: f"<U{frame[col].str.len().max()}"
+                    for col in ent}
+                   for ent, frame in zip(string_cols, (df1, df2))]
+
+    (len_first, col_dtypes,
+     first), (len_last, col_dtypes,
+              last) = list(zip(lengths, string_cols, (df1, df2)))
+
+    #export to numpy as recarray
+    first = first.to_records(column_dtypes=col_dtypes, index=False)
+    #tile first with len_first
+    #remember, len_first is the length of the other dataframe
+    first = np.tile(first, (len_first, 1))
+    #get a 1d array
+    first = np.concatenate(first)
+    #sorting here ensures we get each row of the first
+    #with the entire rows of the other dataframe
+    np.recarray.sort(first, order=first.dtype.names[0])
+
+    #same process as first, except there'll be no sorting
+    last = last.to_records(column_dtypes=col_dtypes, index=False)
+    last = np.tile(last, (len_last, 1))
+    last = np.concatenate(last)
+    result = rfn.merge_arrays((first, last), flatten=True, asrecarray=True)
+    return pd.DataFrame.from_records(result)
+
+
+def grid_computation_list(dfs):
```

Comment:
I believe the same comment as above applies here.

Copy/pasting it for immediate visible context: This function should be imported from the library, rather than re-defined here.